From a5fc04994cc0d858849945801c851cbcadfd896e Mon Sep 17 00:00:00 2001
From: Xogium <contact@xogium.me>
Date: Thu, 7 Jan 2021 14:51:55 -0500
Subject: [PATCH 1/3] stm32mp157c-odyssey: enable usb ports.

This commit enables both usb host ports on board in ehci mode, as well as the OTG (type c) port in ohci mode.
---
 arch/arm/boot/dts/stm32mp157c-odyssey.dts | 33 +++++++++++++++++++++++
 1 file changed, 33 insertions(+)

diff --git a/arch/arm/boot/dts/stm32mp157c-odyssey.dts b/arch/arm/boot/dts/stm32mp157c-odyssey.dts
index 73da983bd76a..4fa0d2349300 100644
--- a/arch/arm/boot/dts/stm32mp157c-odyssey.dts
+++ b/arch/arm/boot/dts/stm32mp157c-odyssey.dts
@@ -136,3 +136,36 @@ &i2c4 {
 &dts {
 	status = "okay";
 };
+
+&usbphyc {
+	status = "okay";
+};
+
+&usbh_ehci {
+	phys = <&usbphyc_port0>, <&usbphyc_port1 1>;
+	phy-names = "usb1-phy", "usb2-phy";
+	status = "okay";
+};
+
+&usbphyc_port0 {
+	phy-supply = <&vdd_usb>;
+	vbus-supply = <&usb1vbus>;
+	vdda1v1-supply = <&reg11>;
+	vdda1v8-supply = <&reg18>;
+};
+
+&usbphyc_port1 {
+	phy-supply = <&vdd_usb>;
+	vbus-supply = <&usb2vbus>;
+	vdda1v1-supply = <&reg11>;
+	vdda1v8-supply = <&reg18>;
+};
+
+&usbotg_hs {
+	compatible = "st,stm32mp15-fsotg", "snps,dwc2";
+	pinctrl-names = "default";
+	pinctrl-0 = <&usbotg_hs_pins_a &usbotg_fs_dp_dm_pins_a>;          /* configure OTG ID and full-speed data pins */
+	vbus-supply = <&vbus_otg>;                                        /* voltage regulator to supply Vbus */
+	dr_mode = "peripheral";
+	status = "okay";
+};
-- 
2.29.2

